mergify_engine/context.py

```diff
@@ -822,9 +823,33 @@ async def _get_consolidated_data(self, name):
                 for ctext, state in (await self.checks).items()
                 if state == "stale"
             ]
+        elif name == "depends-on":
+            # TODO(sileht): This is the list of merged pull request that
+            # required-by this pull request
+            # An optimisation can be to look at the merge queues too, to queue
+            # this pull request earlier
+            depends_on = []
+            for pull_request_number in self.get_depends_on():
+                try:
+                    ctxt = await self.repository.get_pull_request_context(
+                        pull_request_number
+                    )
+                except http.HTTPNotFound:
+                    continue
+                if ctxt.pull["merged"]:
+                    depends_on.append(f"#{pull_request_number}")
+            return depends_on
         else:
             raise PullRequestAttributeError(name)
 
+    DEPENDS_ON = re.compile(r"^ *Depends-On: #(\d+) *$", re.MULTILINE | re.IGNORECASE)
```

Comment:
We should have unit tests for this.

It should also support URL-based pull requests, e.g:
```
Depends-On: https://github.com/myorg/myrepo/pull/123
```